### PR TITLE
fix(client): render top contributor link depending on the language

### LINKED ISSUE #43225: top contributor link in non english profiles broken
The top contributor link in user profiles of non English languages is broken

To reproduce:
go to https://www.freecodecamp.org/espanol/mot01
click the `top contributor` (`Contribuyente principal`) link
page hangs

It tries to take you to https://www.freecodecamp.org/espanol/top-contributors
Chinese goes to https://chinese.freecodecamp.org/top-contributors, which gives a 404

English goes to https://freecodecamp.org/top-contributors, which redirects to the news article

### PR DESCRIPTION
**Pull request by Naufal2511**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [ ] I have tested these changes either locally on my machine, or GitPod.
( I am sorry, I am a beginner and found running the code and testing it really confusing)

Closes #43225

I have tried to follow the instructions, Hope my changes get merged.

Thank You for this opportunity

### CODE REVIEW

File: client/src/components/profile/components/Camper.tsx [hunk 1]
```diff
@@ -129,7 +129,7 @@ function Camper({
           <br />
           <p className='text-center yearsTopContributor'>
             <FontAwesomeIcon icon={faAward} />{' '}
-            <Link to={'/top-contributors'}>{t('profile.contributor')}</Link>
+            <Link {t('links:top-contributors')}={'/top-contributors'}>{t('profile.contributor')}</Link>
```
Comment (nhcarrigan) [inline client/src/components/profile/components/Camper.tsx:132]:
```suggestion
            <Link to={t('links:top-contributors')}>{t('profile.contributor')}</Link>
```

**Review by naomi-lgbt — DISMISSED** [review]
Hey there, thanks for taking care of this.

I've added a suggestion to fix the link element.

File: client/src/components/profile/components/Camper.tsx (hunk 1 above)
Comment (Naufal2511) [reply client/src/components/profile/components/Camper.tsx]: Hey, I made the changes as you had suggested, can you please have a look now.

File: client/src/components/profile/components/Camper.tsx [hunk 2]
```diff
@@ -129,7 +129,7 @@ function Camper({
           <br />
           <p className='text-center yearsTopContributor'>
             <FontAwesomeIcon icon={faAward} />{' '}
-            <Link to={'/top-contributors'}>{t('profile.contributor')}</Link>
+            <Link to={t('links:top-contributors')}>{t('profile.contributor')}</Link>
```
Comment (nhcarrigan) [inline client/src/components/profile/components/Camper.tsx:132]:
```suggestion
            <Link to={t('links:top-contributors')}>
              {t('profile.contributor')}
            </Link>
```
My fault. Forgot about the linter.

Comment (Naufal2511) [reply client/src/components/profile/components/Camper.tsx]: Can You please check now, I have done what you had directed me to do.

**Comment by nhcarrigan** [comment]
This is looking good! It looks like the cypress test needs to be updated, however.

**Comment by nhcarrigan** [comment]
Okay @Naufal2511 to update the test:

In the `/cypress/integration/top-contributor.js` file, change line 35 from `/top-contributor` to `https://freecodecamp.org/top-contributor` and the test should pass.

**Comment by Naufal2511** [comment]
I have changed in /cypress/integration/top-contributor.js as you had mentioned.
Please have a look.
Thank You so much for your patience with me and the proper set of instructions that make things easy for newbies like me.

File: cypress/integration/top-contributor.js [hunk 3]
```diff
@@ -32,7 +32,7 @@ describe('Top contributor in user profile', () => {
     cy.contains('Top Contributor').should(
       'have.attr',
       'href',
-      '/top-contributors'
+      'https://freecodecamp.org/top-contributor'
```
Comment (gikf) [inline cypress/integration/top-contributor.js:35]:
```suggestion
      'https://freecodecamp.org/top-contributors'
```

**Review by gikf — CHANGES_REQUESTED** [review]
This ended up as a more eventful than most of the first issues, :smiley: 

Just one more little change is needed to deal with the cypress test.

**Comment by Naufal2511** [comment]
I have changed top-contributor to top-contributors on line 35.
Please have a look.
And ThankYou again, even though I don't have any idea about what I am doing, I am still getting things done due to all your help.

**Review by gikf — DISMISSED** [review]
Seems to be good :+1:

File: cypress/integration/top-contributor.js (hunk 3 above)
Comment (nhcarrigan) [reply cypress/integration/top-contributor.js]: Whoops my bad 😓

**Comment by Sembauke** [comment]
Hi @Naufal2511, Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to freeCodeCamp 👍
We are happy to accept these changes and look forward to future contributions. 📝

**Review by Sembauke — DISMISSED** [review]

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
I might be misunderstanding our config stuff, but I would assume, seeing as we control this action, we should not link to something requiring a redirect. That is, we should be using the active link.

_Open for discussion/correction_

File: client/i18n/locales/chinese-traditional/links.json [hunk 4]
```diff
@@ -1,5 +1,6 @@
 {
   "help-translate-url": "https://contribute.freecodecamp.org/#/i18n/chinese/how-to-translate-files",
+  "top-contributors": "https://freecodecamp.org/top-contributors",
```
Comment (ojeytonwilliams) [inline client/i18n/locales/chinese-traditional/links.json:3]:
```suggestion
  "top-contributors": "https://www.freecodecamp.org/news/freecodecamp-top-contributors/",
```

File: client/i18n/locales/chinese/links.json [hunk 5]
```diff
@@ -1,5 +1,6 @@
 {
   "help-translate-url": "https://contribute.freecodecamp.org/#/i18n/chinese/how-to-translate-files",
+  "top-contributors": "https://freecodecamp.org/top-contributors",
```
Comment (ojeytonwilliams) [inline client/i18n/locales/chinese/links.json:3]:
```suggestion
  "top-contributors": "https://www.freecodecamp.org/news/freecodecamp-top-contributors/",
```

File: client/i18n/locales/english/links.json [hunk 6]
```diff
@@ -1,5 +1,6 @@
 {
   "help-translate-link-url": "https://contribute.freecodecamp.org/#/how-to-translate-files",
+  "top-contributors": "https://freecodecamp.org/top-contributors",
```
Comment (ojeytonwilliams) [inline client/i18n/locales/english/links.json:3]:
```suggestion
  "top-contributors": "https://www.freecodecamp.org/news/freecodecamp-top-contributors/",
```

File: client/i18n/locales/espanol/links.json [hunk 7]
```diff
@@ -1,5 +1,6 @@
 {
   "help-translate-link-url": "https://contribute.freecodecamp.org/#/i18n/espanol/how-to-translate-files",
+  "top-contributors": "https://freecodecamp.org/top-contributors",
```
Comment (ojeytonwilliams) [inline client/i18n/locales/espanol/links.json:3]:
```suggestion
  "top-contributors": "https://www.freecodecamp.org/news/freecodecamp-top-contributors/",
```

File: client/i18n/locales/italian/links.json [hunk 8]
```diff
@@ -1,5 +1,6 @@
 {
   "help-translate-link-url": "https://contribute.freecodecamp.org/#/i18n/italian/how-to-translate-files",
+  "top-contributors": "https://freecodecamp.org/top-contributors",
```
Comment (ojeytonwilliams) [inline client/i18n/locales/italian/links.json:3]:
```suggestion
  "top-contributors": "https://www.freecodecamp.org/news/freecodecamp-top-contributors/",
```

File: client/i18n/locales/portuguese/links.json [hunk 9]
```diff
@@ -1,5 +1,6 @@
 {
   "help-translate-link-url": "https://contribute.freecodecamp.org/#/i18n/portuguese/how-to-translate-files",
+  "top-contributors": "https://freecodecamp.org/top-contributors",
```
Comment (ojeytonwilliams) [inline client/i18n/locales/portuguese/links.json:3]:
```suggestion
  "top-contributors": "https://www.freecodecamp.org/news/freecodecamp-top-contributors/",
```

File: cypress/integration/top-contributor.js [hunk 10]
```diff
@@ -32,7 +32,7 @@ describe('Top contributor in user profile', () => {
     cy.contains('Top Contributor').should(
       'have.attr',
       'href',
-      '/top-contributors'
+      'https://freecodecamp.org/top-contributors'
```
Comment (ojeytonwilliams) [inline cypress/integration/top-contributor.js:35]:
```suggestion
      'https://www.freecodecamp.org/news/freecodecamp-top-contributors/'
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
I agree with Shaun, we should link directly to the top contributor news article, rather than rely on the redirection.  The point of the redirection is to make sure that inbound links keep working, not to be the ultimate source of truth.

@Naufal2511 sorry for requesting yet another set of changes!

**Comment by Naufal2511** [comment]
I have changed the above-mentioned mentions.
I am actually enjoying this process, so if there are any other issues I can work on please direct me to them.
I am a amateur at C++, C and XAMPP stack (HTML CSS JS PHP MySQL)

**Review by ojeytonwilliams — DISMISSED** [review]
LGTM :+1: Well, it looks good aside from the conflicts!

**Comment by ieahleen** [comment]
As this was your first contribution, I have resolved the merge issue.

**Review by ojeytonwilliams — APPROVED** [review]

**Comment by Naufal2511** [comment]
Thanks

**Review by ShaunSHamilton — APPROVED** [review]
This LGTM 👍

**Pull request merged**